cpp/util/etcd.cc

```diff
@@ -581,8 +588,11 @@ void EtcdClient::WatchInitialGetAllDone(WatchState* state, util::Status status,
 
   state->known_keys_.swap(new_known_keys);
 
-  state->task_->executor()->Add(
-      bind(&EtcdClient::SendWatchUpdates, this, state, move(updates)));
+  if (!updates.empty() || state->first_update_) {
+    state->first_update_ = false;
+    state->task_->executor()->Add(
+        bind(&EtcdClient::SendWatchUpdates, this, state, move(updates)));
```

Comment:
Can't do this! ```SendWatchUpdates``` sends the update... *and* starts the next request!

I was going to point out that there's two spots calling it, as well, so if you're going to do it this way, should do it centrally, in ```SendWatchUpdates```, but as it turns out, it's the only way to go.